scRNA-seq/setup/mouse-liver/prepare_mouse_liver_markers.Rmd

```diff
@@ -0,0 +1,326 @@
+colnames(sce) <- colData(sce)$CellID
+```
+
+### Filtering cells
+
+There are already QC metrics associated with this object ([docs](https://data.humancellatlas.org/pipelines/qc-metrics))
+
+```{r}
+names(colData(sce))
+```
+
+First, we'll filter on the basis of the emptyDrops FDR.
+
+```{r}
+filtered_sce <- sce[, sce$emptydrops_FDR <= 0.01]
+```
+
+Then on the basis of % mitochondrial molecules.
+
+```{r}
+filtered_sce <- filtered_sce[, filtered_sce$pct_mitochondrial_molecules < 10] 
+```
+
+We will get a few negative size factors if we don't remove cells with few expressed features.
+We'll filter on `n_genes` to account for this.
+
+```{r}
+filtered_sce <- filtered_sce[, filtered_sce$n_genes > 200]
+```
+
+### Filtering genes
+
+```{r}
+filtered_sce <- scater::addPerFeatureQC(filtered_sce)
+# Genes detected more than 3% of cells
+detected <- rowData(filtered_sce)$detected > 3
+# Genes with mean count greater than 0.1
+expressed <- rowData(filtered_sce)$mean > 0.1
+
+filtered_sce <- filtered_sce[detected & expressed, ]
+```
+
+What did we retain with these cutoffs?
+
+```{r}
+dim(filtered_sce)
+```
+
+### Normalization
+
+```{r normalize}
+# Set seed
+set.seed(2021)
+
+# Cluster similar cells
+qclust <- scran::quickCluster(filtered_sce)
+
+# Compute sum factors for each cell cluster grouping.  
+filtered_sce <- scran::computeSumFactors(filtered_sce, 
+                                         clusters = qclust,
+                                         # Setting this to FALSE allows us to
+                                         # do our own diagnostics if we get
+                                         # negative size factors
+                                         positive = FALSE)
+```
+
+Any negative size factors?
+
+```{r}
+any(filtered_sce$sizeFactor < 0)
+```
+
+Great, let's normalize and transform our data.
+
+```{r}
+# Normalize and log transform. 
+normalized_sce <- scater::logNormCounts(filtered_sce)
+```
+
+Remove all the but the normalized `SingleCellExperiment`.
+
+```{r}
+rm(sce, filtered_sce)
+```
+
+
+## Clustering
+
+### HVGs
+
+Feature selection, e.g., identify high variance genes
+
+```{r}
+# Model gene variance
+gene_variance <- scran::modelGeneVar(normalized_sce)
+# Select the most variable genes
+highvar_genes <- scran::getTopHVGs(gene_variance, n = 2500)
+```
+
+### PCA
+
+Perform PCA on the high variance genes
+
+```{r}
+# Calculate and store PC matrix
+normalized_sce <- scater::runPCA(normalized_sce, subset_row = highvar_genes)
+```
+
+Plot PC1 and PC2
+
+```{r}
+scater::plotReducedDim(normalized_sce, "PCA")
+```
+
+### Clusters
+
+We'll perform clustering on the PCA matrix.
+
+```{r}
+# Extract the principal components matrix
+pca_mat <- reducedDim(normalized_sce, "PCA")
+```
+
+Cluster the cells in PCA space, using graph-based clustering.
+
+```{r}
+# Run the clustering algorithm
+nnclusters <- bluster::clusterRows(
+  pca_mat, 
+  bluster::NNGraphParam(k = 10, 
+                        cluster.fun = "louvain")
```

Comment:
Just wondering about parameter choice here... Is there a reason not to just use defaults? If not, was there any reason to use louvain w/o jaccard? The only reason I ask is b/c jaccard/louvain is the seurat default, so they are in my mind together.

Reply:
I don't have a good reason. For now, I've removed this change - this will just be setting `k = 10`.

Reply:
Just an update here - changing these parameters was not a good idea in terms of the required downstream changes cost vs. the benefit (this makes sense and also you said that at the outset!) so will be rolling that back.